src/components/AccountSettings.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect } from 'react'
+import { Button, Col, Container, FormControl, FormGroup, FormLabel, Row } from 'react-bootstrap'
+import { useSynapseContext } from 'synapse-react-client/dist/utils/SynapseContext'
+import { UserProfile, UserBundle } from 'synapse-react-client/dist/utils/synapseTypes'
+import { SynapseConstants } from 'synapse-react-client'
+import { getMyUserBundle, updateMyUserProfile } from 'synapse-react-client/dist/utils/SynapseClient'
+import { displayToast } from 'synapse-react-client/dist/containers/ToastMessage'
+import StarterAccount from '../assets/StarterAccount.svg'
+import VerifedAccount from '../assets/VerifiedAccount.svg'
+import EditIcon from '../assets/RedEditPencil.svg'
+import ChangePasswordPage from './ChangePassword'
+import { ORCiDButton } from './ORCiDButton'
+export type AccountSettingsProps = {
+}
+
+const AccountSettings = (props: AccountSettingsProps) => {
+    const { accessToken } = useSynapseContext()
+    const [ userProfile, setUserProfile ] = useState<UserProfile>()
+    const [ orcid, setOrcid ] = useState<string>()
+    const [ verified, setVerfied ] = useState<boolean>()
+    const [ editUsername, setEditUsername] = useState<boolean>()
+    const [ editEmail, setEditEmail] = useState<boolean>()
+    const [ changePW, setChangePW] = useState<boolean>()
+    const [ updatedUsername, setUpdatedUsername] = useState<string>('')
+    const [ updatedEmail, setUpdatedEmail] = useState<string>('')
+
+    const onUpdateUserProfile = async (event: React.SyntheticEvent) => {
+        event.preventDefault()
+        try{
+            if(userProfile && editUsername && updatedUsername){
+                userProfile.userName = updatedUsername
+                const updatedProfile = await updateMyUserProfile(userProfile, accessToken)
+                setUserProfile(updatedProfile)
+            }
+
+            // Placeholder for updating emails
+            // if(userProfile && editEmail && updatedEmail){
+            //     userProfile.emails = [...userProfile.emails!, updatedEmail]
```

Comment:
You can remove the user profile based email update code, since (as you found out) that's not how it's done.